internal/resources/cluster/data_source_cluster.go

```diff
@@ -45,6 +47,11 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 		if err != nil {
 			if clienterrors.IsNotFoundError(err) {
 				_ = schema.RemoveFromState(d, m)
+
+				if ctx.Value(contextMethodKey{}) == "read" {
```

Comment:
you are erroring out after `_ = schema.RemoveFromState(d, m)`.. this is wrong
You should not remove it from the state and just error out in the case read context

Reply:
Done